WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanFragment.kt

```diff
@@ -3,26 +3,68 @@ package org.wordpress.android.ui.jetpack.scan
 import android.os.Bundle
 import android.view.View
 import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
 import androidx.lifecycle.ViewModelProvider
+import kotlinx.android.synthetic.main.scan_fragment.recycler_view
 import org.wordpress.android.R
 import org.wordpress.android.WordPress
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.ui.jetpack.scan.ScanViewModel.UiState.Content
+import org.wordpress.android.ui.jetpack.scan.adapters.ScanAdapter
+import org.wordpress.android.ui.utils.UiHelpers
 import javax.inject.Inject
 
 class ScanFragment : Fragment(R.layout.scan_fragment) {
+    @Inject lateinit var uiHelpers: UiHelpers
     @Inject lateinit var viewModelFactory: ViewModelProvider.Factory
     private lateinit var viewModel: ScanViewModel
 
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-        (requireActivity().application as WordPress).component().inject(this)
-    }
-
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        initViewModel()
+
+        val nonNullActivity = requireActivity()
+        (nonNullActivity.application as WordPress).component()?.inject(this)
+
+        val site = if (savedInstanceState == null) {
+            val nonNullIntent = checkNotNull(nonNullActivity.intent)
+            nonNullIntent.getSerializableExtra(WordPress.SITE) as SiteModel
+        } else {
+            savedInstanceState.getSerializable(WordPress.SITE) as SiteModel
+        }
+
+        initAdapter()
+        initViewModel(site)
+    }
+
+    private fun initAdapter() {
+        val scanAdapter = ScanAdapter(uiHelpers)
+        scanAdapter.setHasStableIds(true)
+        recycler_view.adapter = scanAdapter
     }
 
-    private fun initViewModel() {
+    private fun initViewModel(site: SiteModel) {
         viewModel = ViewModelProvider(this, viewModelFactory).get(ScanViewModel::class.java)
+        setupObservers()
+        viewModel.start(site)
+    }
+
+    private fun setupObservers() {
+        viewModel.uiState.observe(
+            viewLifecycleOwner,
+            Observer { uiState ->
```

Comment:
Question (❓ ): Does your IDE also suggests that this redundant SAM-constructor can be removed?
Minor (🔍 ): In the end I can make this code look like the below:
```
        viewModel.uiState.observe(viewLifecycleOwner, { uiState ->
            if (uiState is Content) refreshContentScreen(uiState)
        })
```

PS: Usually, in Kotlin, I tend to keep simple, not complex, if statements, a one-liner. Thus, the suggestion above. If I can recall correctly this is also something that is suggested in the official Android Kotlin style guide.

Reply:
> Does your IDE also suggests that this redundant SAM-constructor can be removed?

My IDE highlights it as well. It is for [this reason](https://github.com/wordpress-mobile/WordPress-Android/pull/13469#discussion_r530843257) mentioned in another PR review that I included it:
> we're not removing them yet in the code for compatibility with older AS versions, right?

----
> I tend to keep simple, not complex, if statements, a one-liner.

I just followed existing code conventions in the VMs which uses braces around `if`:

```
if (isStarted) {
    return
}
```

Reply:
> My IDE highlights it as well. It is for this reason mentioned in another PR review that I included it:

Interesting, thanks for sharing Ashita, I have left the below comment there and quoting it here.

> Hmmm... Is this due to compatibility reason with AS version or Kotlin versions? I am thinking the later, right? And if that is the case, I suggest having every develop be forced to update the AS Kotlin plugin to at least the Gradle's Kotlin version, currently 1.4.10, as specified by our build files. This will make sure everyone is on the same boat. Wdyt?